code/go/0chain.net/chaincore/chain/handler.go

```diff
@@ -1681,14 +1681,19 @@ func (c *Chain) MinerStatsHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func txnIterHandlerFunc(c *Chain, w http.ResponseWriter, s *state.State) func(context.Context, datastore.CollectionEntity) (bool, error) {
+func txnIterHandlerFunc(c *Chain, w http.ResponseWriter) func(context.Context, datastore.CollectionEntity) (bool, error) {
 	return func(ctx context.Context, ce datastore.CollectionEntity) (bool, error) {
 		txn, ok := ce.(*transaction.Transaction)
 		if !ok {
 			logging.Logger.Error("generate block (invalid entity)", zap.Any("entity", ce))
 			return false, nil
 		}
 
+		s, err := c.GetStateById(c.GetLatestFinalizedBlock().ClientState, txn.ClientID)
```

Comment:
We'd better create a `balances` or stateContext from `LFB.ClientState` outside the iteration, and get client state from it. The `c.GetLatestFinalizedBlock()` could change while iterating.